src/apps/search-header/search-header.tsx

```diff
@@ -252,8 +252,9 @@ const SearchHeader: React.FC = () => {
             searchUrl,
             selectedItemString,
             filter: {
-              materialType:
+              materialType: encodeURI(
                 autosuggestCategoryList[highlightedCategoryIndex].type
+              )
```

Comment:
Please move this encoding to `constructSearchUrlWithFilter()`.

This way we avoid forcing each caller to deal with encoding.

Based on the documentation it also seems as if [`encodeURI()` deals with an entire URI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI) - not just a single part. This also points to the fact that this responsibility should be push down the call stack where we have the entire URL available.

Reply:
good point! I moved it into the `appendQueryParametersToUrl()` function, where it all goes down.